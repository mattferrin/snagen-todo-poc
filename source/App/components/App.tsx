import React, { FC } from "react";
import { Text } from "ink";

const App: FC<{ readonly name?: string }> = ({ name = "Stranger" }) => (
  <Text>
    Hello, <Text color="green">{name}</Text>
  </Text>
);

export default App;
