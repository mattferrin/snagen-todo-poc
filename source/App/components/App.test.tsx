import ShallowRenderer from "react-test-renderer/shallow";
import React from "react";
import App from "./App";

type SnapInput = {
  readonly stabileSortId: string;
  readonly comment: string | null;
  readonly props: Parameters<typeof App>[0];
};

function snap(input: SnapInput): void {
  it(input.stabileSortId, () => {
    /** setup mocks */

    /** unit under test */
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<App {...input.props} />);
    const result = renderer.getRenderOutput();

    /** assertions */
    const comment = `${input.comment === "" ? "TODO: comment" : input.comment}`;
    expect({
      [comment]: { result },
    }).toMatchSnapshot();
  });
}

snap({
  stabileSortId: "a6fb76eb-b9bb-4711-bcbe-2168031eecf4",
  comment: "proved jest is working",
  props: { name: "Matt" },
});
