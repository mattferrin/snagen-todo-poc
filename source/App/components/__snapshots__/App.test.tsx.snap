// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a6fb76eb-b9bb-4711-bcbe-2168031eecf4 1`] = `
{
  "proved jest is working": {
    "result": <Text
      bold={false}
      dimColor={false}
      italic={false}
      strikethrough={false}
      underline={false}
      wrap="wrap"
    >
      Hello, 
      <Text
        bold={false}
        color="green"
        dimColor={false}
        italic={false}
        strikethrough={false}
        underline={false}
        wrap="wrap"
      >
        Matt
      </Text>
    </Text>,
  },
}
`;
